datafusion/sql/src/expr/mod.rs

```diff
@@ -291,7 +291,9 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                 )))
             }
 
-            SQLExpr::TypedString { data_type, value } => Ok(Expr::Cast(Cast::new(
+            SQLExpr::TypedString(TypedString {
+                data_type, value, ..
```

Comment:
Would it be better to avoid the `..` and explicitly handle the new field? I think it's good to get compiler errors if new fields are added that we need to consider.